config/stored_requests.go

```diff
@@ -318,37 +432,43 @@ type PostgresUpdatePolling struct {
 	AmpQuery string `mapstructure:"amp_query"`
 }
 
-func (cfg *PostgresUpdatePolling) validate(dataType DataType, errs []error) []error {
+func (cfg *DatabaseUpdatePolling) validate(dataType DataType, wildcard string, errs []error) []error {
 	section := dataType.Section()
 	if cfg.Query == "" {
 		return errs
 	}
 
 	if cfg.RefreshRate <= 0 {
-		errs = append(errs, fmt.Errorf("%s: postgres.poll_for_updates.refresh_rate_seconds must be > 0", section))
+		errs = append(errs, fmt.Errorf("%s: database.poll_for_updates.refresh_rate_seconds must be > 0", section))
 	}
 
 	if cfg.Timeout <= 0 {
-		errs = append(errs, fmt.Errorf("%s: postgres.poll_for_updates.timeout_ms must be > 0", section))
+		errs = append(errs, fmt.Errorf("%s: database.poll_for_updates.timeout_ms must be > 0", section))
 	}
-
-	if !strings.Contains(cfg.Query, "$1") || strings.Contains(cfg.Query, "$2") {
-		errs = append(errs, fmt.Errorf("%s: postgres.poll_for_updates.query must contain exactly one wildcard", section))
+	if wildcard == "?" {
+		if strings.Count(cfg.Query, "?") != 2 {
+			errs = append(errs, fmt.Errorf("%s: database.poll_for_updates.query must contain exactly two wildcards", section))
+		}
+	}
+	if wildcard == "$" {
+		if !strings.Contains(cfg.Query, "$1") || strings.Contains(cfg.Query, "$3") {
+			errs = append(errs, fmt.Errorf("%s: database.poll_for_updates.query must contain exactly one wildcard", section))
+		}
```

Comment:
This validation, specifically line 449 leads to error with a valid sql query. I don't think this condition should exist. For example my `poll_for_updates.query` looks like 
```
"SELECT uuid AS id, config as data, 'request' as dataType FROM videoconfig_config WHERE updated_at > ?"
```

Reply:
The original validation at line #454 (for postgres) makes sure that the query has exactly `one` parameter.
The validation at line #449 does the same but for mysql but in this case the query should have `two` parameters.
See my notes [here](https://github.com/prebid/prebid-server/pull/2356/files/6343c15e6031d7465857269e3c67922aeb7d6b62#r969375457)

I see here a mistake. Line #454 should be:
```
if !strings.Contains(cfg.Query, "$1") || strings.Contains(cfg.Query, "$2") {
```
I will change it and commit.

Reply:
Agree, thank you for this fix